tests/functional/h/views/admin/test_permissions.py

```diff
@@ -0,0 +1,70 @@
+import pytest
+
+
+class TestAdminPermissions:
+    PAGES = (
+        # URL, accessible by role.Staff?
+        ("/admin/", True),
+        ("/admin/admins", False),
+        ("/admin/badge", False),
+        ("/admin/features", False),
+        ("/admin/groups", True),
+        ("/admin/mailer", True),
+        ("/admin/nipsa", False),
+        ("/admin/oauthclients", False),
+        ("/admin/organizations", True),
+        ("/admin/staff", False),
+        ("/admin/users", True),
+        ("/admin/search", False),
+    )
+
+    @pytest.mark.parametrize("url", (page[0] for page in PAGES))
+    def test_not_accessible_by_regular_user(self, app, url, user):
+        self.login(app, user)
+
+        res = app.get(url, expect_errors=True)
+
+        assert res.status_code == 404
```

Comment:
I think you can just do:

```suggestion
        app.get(url, status=404)
```

See the `status` arg to `get()`: http://webtest.pythonpaste.org/en/latest/api.html#webtest.app.TestApp.get